lib/ace/ext/papeeria/spellchecker.coffee

```diff
@@ -37,196 +35,206 @@ define((require, exports, module) ->
+    # Bindings for editor keys with PopupManager functions.
+    commands:
+      "Esc": ->
+        PopupManager.detach()
+      "Up": ->
+        PopupManager.goTo("up")
+      "Down": ->
+        PopupManager.goTo("down")
+      "Return": ->
+        PopupManager.insertCorrection()
+
+    # Initial setup for popup.
+    # @param {Editor} editor: editor object.
+    init: (editor) ->
+      @editor = editor
+
+      HashHandler = require("ace/keyboard/hash_handler").HashHandler
+      AcePopup = require("ace/autocomplete/popup").AcePopup
+
+      @popup = new AcePopup(document.body)
+      @popup.setTheme(editor.getTheme())
+      @popup.setFontSize(editor.getFontSize())
+      @popup.on("click", (e) =>
+        @insertCorrection()
+        e.stop()
+      )
+
+      @session = editor.session
+
+      @keyboard = new HashHandler()
+      @keyboard.bindKeys(@commands)
+
+      @detach if editor?
+      return
+
+    # Show popup in editor.
+    # @param {Array} options: list of corrections for the current word.
+    show: (options) ->
+      @options = options
+
+      @popup.setData(options)
+
+      @editor.keyBinding.addKeyboardHandler(@keyboard)
+
+      position = @editor.getCursorPosition()
+      @base = @session.doc.createAnchor(position.row, position.column)
+
+      PopupManager.detach if editor?
+      @editor.on("change", @changeListener)
+      @editor.on("changeSelection", @changeSelectionListener)
+      @editor.on("blur", @blurListener)
+      @editor.on("mousedown", @mousedownListener)
+      @editor.on("mousewheel", @mousewheelListener)
+      @editor.$blockScrolling = Infinity
+
+      @activated = true
+
+      renderer = @editor.renderer
+      lineHeight = renderer.layerConfig.lineHeight
+      pos = renderer.$cursorLayer.getPixelPosition(this.base, true)
+      pos.left -= @popup.getTextLeftOffset()
+      rect = @editor.container.getBoundingClientRect()
+      pos.top += rect.top - renderer.layerConfig.offset
+      pos.left += rect.left - @editor.renderer.scrollLeft
+      pos.left += renderer.gutterWidth
+
+      @popup.show(pos, lineHeight)
+      return
+
+    # Detach popup from editor.
+    detach: ->
+      @popup.hide() if @popup and @popup.isOpen
+
+      @base.detach() if @base
+
+      @editor.keyBinding.removeKeyboardHandler(@keyboard)
+
+      @editor.off("change", @changeListener)
+      @editor.off("changeSelection", @changeSelectionListener)
+      @editor.off("blur", @blurListener)
+      @editor.off("mousedown", @mousedownListener)
+      @editor.off("mousewheel", @mousewheelListener)
+
+      @activated = false
+      return
+
+    # Insert selected correction (option) instead of the current word.
+    insertCorrection: () ->
+      PopupManager.detach()
+      row = PopupManager.popup.getRow()
+      correction = PopupManager.options[row].value
+      wordRange = getCurrentWordRange(PopupManager.editor)
+      word = PopupManager.editor.session.getTextRange(wordRange)
+      if ' ' in word
```

Comment:
Is it what will become unnecessary when you call getWordRange instead of getAWordRange?